staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/plugin.go

```diff
@@ -44,7 +45,7 @@ const (
 
 var (
 	compositionEnvTemplate *cel.CompositionEnv = func() *cel.CompositionEnv {
```

Comment:
this function is being invoked at package initialization time, before feature gates have been set ... the feature gate check won't be effective.

We may need a strict and non-strict package var, and the place where we're using compositionEnvTemplate has to select which one to use based on the gate state

That said, I don't really understand how the strictCost setting in the env held by `compositionEnvTemplate` interacts with the `StrictCost` setting in the `OptionalVariableDeclarations` structs below passed to filterCompiler.CompileAndStoreVariables / filterCompiler.Compile

if compositionEnvTemplate has a non-strict base env (which it currently does because this is getting initialized at package init time) but filterCompiler.CompileAndStoreVariables / filterCompiler.Compile get passed OptionalVariableDeclarations with StrictCost=true, what is the result?

Reply:
Added more tests. The compositionEnvTemplate initialization here sounds not affecting the underlying env be used since we build all combination of envs and use based on `OptionalVariableDeclarations`. I have updated the pr to keep the feature gate value synced.